cmd/uplink/cmd/cp_test.go

```diff
@@ -4,122 +4,180 @@
 package cmd_test
 
 import (
-	"crypto/rand"
+	"encoding/hex"
+	"errors"
 	"log"
+	"math/rand"
 	"os"
-	"strconv"
 	"testing"
 
 	"storj.io/storj/internal/memory"
 	"storj.io/storj/internal/s3client"
 )
 
 const (
-	bucket                = "testbucket"
-	uplinkEncryptionKey   = "supersecretkey"
-	storjSimSatelliteAddr = "127.0.0.1:10000"
+	bucket = "testbucket"
 )
 
-func setup() s3client.Client {
-	var conf s3client.Config
-	conf.Satellite = storjSimSatelliteAddr
-	conf.EncryptionKey = uplinkEncryptionKey
-	conf.APIKey = os.Getenv("storjSimApiKey")
+var benchmarkCases = []struct {
+	name       string
+	objectsize memory.Size
+}{
+	{"100B", 100 * memory.B},
+	{"1MB", 1 * memory.MiB},
+	{"10MB", 10 * memory.MiB},
+	{"100MB", 100 * memory.MiB},
+	{"1G", 1 * memory.GiB},
+}
+
+var testObjects = createObjects()
+
+// createObjects generates the objects (i.e. slice of bytes) that
+// will be used as the objects to upload/download tests
+func createObjects() map[string][]byte {
+	objects := make(map[string][]byte)
+	for _, bm := range benchmarkCases {
+		data := make([]byte, bm.objectsize)
+		_, err := rand.Read(data)
+		if err != nil {
+			log.Fatalf("failed to read random bytes: %+v\n", err)
+		}
+		objects[bm.name] = data
+	}
+	return objects
+}
 
+// uplinkSetup setups an uplink to use for testing uploads/downloads
+func uplinkSetup() s3client.Client {
+	conf, err := setupConfig()
+	if err != nil {
+		log.Fatalf("failed to setup s3client config: %+v\n", err)
+	}
 	client, err := s3client.NewUplink(conf)
 	if err != nil {
-		log.Fatalf("failed to create NewUplink: %+v\n", err)
+		log.Fatalf("failed to create s3client NewUplink: %+v\n", err)
 	}
 	err = client.MakeBucket(bucket, "")
 	if err != nil {
-		log.Fatalf("failed to create bucket %q: %+v\n", bucket, err)
+		log.Fatalf("failed to create bucket with s3client %q: %+v\n", bucket, err)
 	}
 	return client
 }
 
-func teardown(client s3client.Client) {
-	for _, bm := range benchmarkCases {
-		deleteUploadedFiles(bm.name)
-	}
-
-	err := client.RemoveBucket(bucket)
-	if err != nil {
-		log.Fatalf("failed to remove bucket %q: %+v\n", bucket, err)
+func setupConfig() (s3client.Config, error) {
+	const (
+		uplinkEncryptionKey  = "supersecretkey"
+		defaultSatelliteAddr = "127.0.0.1:10000"
+	)
+	var conf s3client.Config
+	conf.EncryptionKey = uplinkEncryptionKey
+	conf.Satellite = getEnvOrDefault("SATELLITE_0_ADDR", defaultSatelliteAddr)
+	conf.APIKey = getEnvOrDefault(os.Getenv("GATEWAY_0_API_KEY"), os.Getenv("apiKey"))
```

Comment:
Did you mean to have the first `os.Getenv()` call here? If I'm understanding right, I think you just want to pass `"GATEWAY_0_API_KEY"` itself to `getEnvOrDefault()`.

Reply:
oops just saw this comment. i already merged, but i will fix in another PR